File: src/java/foundgine-core/src/main/java/com/foundgine/core/execution/IProviderPlanCompiler.java
package com.foundgine.core.execution;

/**
 * Compiles provider-neutral execution work into a provider-specific plan.
 * Providers consume ExecutionIR only; semantic planning artifacts never cross
 * this boundary.
 */
public interface IProviderPlanCompiler {
    ProviderPlan compile(ExecutionIR ir);
}

File: src/csharp/Foundgine.Core/Execution/IProviderPlanCompiler.cs
namespace Foundgine.Core.Execution;

/// <summary>
/// Compiles provider-neutral execution work into a provider-specific plan.
/// Providers consume ExecutionIR only; semantic planning artifacts never cross
/// this boundary.
/// </summary>
public interface IProviderPlanCompiler
{
    ProviderPlan Compile(ExecutionIR ir);
}
